nototools/android_patches.py

```diff
@@ -32,49 +32,55 @@
 SRC_DIR = tool_utils.resolve_path('[tools]/packages/android')
 DST_DIR = tool_utils.resolve_path('[tools]/packages/android-patched')
 
-def _patch_hyphen_armenian_ethiopic():
-  """Add hyphen-minus glyphs to Armenian and Ethiopic fonts.
+def _patch_hyphen():
+  """Add hyphen-minus glyphs to fonts that need it.
 
-  This is to enable Armenian and Amharic langauges to be hyphenated
-  properly, since Minikin's itemizer currently shows tofus if a an
-  automatically hyphenated word is displated in a font that has neither
-  HYPHEN nor HYPHEN-MINUS.
+  This is to enable languages to be hyphenated properly,
+  since Minikin's itemizer currently shows tofus if an
+  automatically hyphenated word is displated in a font
+  that has neither HYPHEN nor HYPHEN-MINUS.
 
-  In practice only U+002D HYPHEN-MINUS is added, since Noto LGC fonts
+  The list of font names comes from LANG_TO_SCRIPT in
+  tools/font/fontchain_lint.py.
+
+  (In practice only U+002D HYPHEN-MINUS is added, since Noto LGC fonts
   don't have U+2010 HYPHEN.)
 
   Bug: 21570828"""
 
-  FONTS = [
-      'NotoSansArmenian-Regular.ttf',
-      'NotoSansArmenian-Bold.ttf',
-      'NotoSerifArmenian-Regular.ttf',
-      'NotoSerifArmenian-Bold.ttf',
-      'NotoSansEthiopic-Regular.ttf',
-      'NotoSansEthiopic-Bold.ttf',
-  ]
+  # Names of fonts for which Android requires a hyphen.
+  # This list omits Japanese and Korean.
+  script_names = [
+      'Armenian', 'Ethiopic', 'Bengali', 'Gujarati', 'Devanagari',
+      'Kannada', 'Malayalam', 'Oriya', 'Gurmukhi', 'Tamil', 'Telugu']
 
```

Comment:
Were you going to update the LGC fonts with the prime glyphs from the fonts
Marek made, or should I?

On Tue, Mar 22, 2016 at 11:39 PM, jungshik <notifications@github.com> wrote:

> In nototools/android_patches.py
> <https://github.com/googlei18n/nototools/pull/204#discussion_r57115785>:
>
> >    don't have U+2010 HYPHEN.)
> >
> >    Bug: 21570828"""
> >
> > -  FONTS = [
> > -      'NotoSansArmenian-Regular.ttf',
> > -      'NotoSansArmenian-Bold.ttf',
> > -      'NotoSerifArmenian-Regular.ttf',
> > -      'NotoSerifArmenian-Bold.ttf',
> > -      'NotoSansEthiopic-Regular.ttf',
> > -      'NotoSansEthiopic-Bold.ttf',
> > -  ]
> > +  # Names of fonts for which Android requires a hyphen.
> > +  # This list omits Japanese and Korean.
> > +  script_names = [
> > +      'Armenian', 'Ethiopic', 'Bengali', 'Gujarati', 'Devanagari',
> > +      'Kannada', 'Malayalam', 'Oriya', 'Gurmukhi', 'Tamil', 'Telugu']
> >
>
> Yup, these are the list (as I mentioned in an internal CL). Anyway,
> because it turned out that fonts for newly added scripts already have
> hyphen, I can go ahead with my change internally without having to wait for
> this (+ font update).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/googlei18n/nototools/pull/204/files/91e448dd04a0597555c280ff1d3a572ea3fdd96d..3abfb83c4209906fa5c43933954c2bdfe4e7ba53#r57115785>
>